dashboard/app/models/course.rb

```diff
@@ -242,6 +242,16 @@ def self.valid_course_id?(course_id)
     valid_courses.any? {|course| course[:id] == course_id.to_i}
   end
 
+  # @param user [User]
+  # @param course_id [String] id of the course we're checking
+  # @returns [Boolean] Whether the user can assign this course.
+  # Users should only be able to assign one of their valid courses.
+  def assignable?(user, course_id)
```

Comment:
since `assignable?` is an instance method, we have access to `id` within this method automatically. that means we can remove the `course_id` param here and change line 251 to:

```
Course.valid_course_id?(id)
```

and we can make the same change in the `script.assignable?` method